packages/playwright-test/src/reporters/azure.ts

```diff
@@ -0,0 +1,481 @@
+      `${this.environment ? `[${this.environment}]:` : ''}Test plan ${this.planId}`;
+    this.uploadAttachments = this._options.uploadAttachments || false;
+    this.attachmentsType = this._options.attachmentsType;
+    this.token = this._options.token;
+
+    if (!this.orgUrl || this.orgUrl.length === 0) {
+      this.log(colors.yellow("'orgUrl' is not set. Reporting is disabled."));
+      this.isDisabled = true;
+      return;
+    }
+    if (!this.projectName || this.projectName.length === 0) {
+      this.log(colors.yellow("'projectName' is not set. Reporting is disabled."));
+      this.isDisabled = true;
+      return;
+    }
+    if (!this.planId || this.planId === 0) {
+      this.log(colors.yellow("'planId' is not set. Reporting is disabled."));
+      this.isDisabled = true;
+      return;
+    }
+    if (!this.token || this.token.length === 0) {
+      this.log(colors.yellow("'token' is not set. Reporting is disabled."));
+      this.isDisabled = true;
+    }
+    if (this.uploadAttachments) {
+      if (!this.attachmentsType) {
+        this.log(colors.yellow("'attachmentsType' is not set. Attachments Type will be set to 'screenshot' by default."));
+        this.attachmentsType = ['screenshot'];
+      }
+    }
+    this.connection = new azdev.WebApi(this.orgUrl, azdev.getPersonalAccessTokenHandler(this.token));
+  }
+
+  async onBegin(): Promise<void> {
+    if (this.isDisabled)
+      return;
+
+    this.testApi = await this.connection.getTestApi();
+
+    this.createRun(this.testRunTitle).then(run => {
+      if (run) {
+        this.runId = run.id;
+        this.log(colors.green(`Using run ${this.runId} to publish test results`));
+      } else {
+        this.isDisabled = true;
+      }
+    });
+  }
+
+  async onTestEnd(test: TestCase, testResult: TestResult): Promise<void> {
+    await this.awaitForRunId().catch(e => {
+      this.log(colors.red(e));
+    });
+    if (this.isDisabled)
+      return;
+
+    this.logTestItem(test, testResult);
+    await this.publishCaseResult(test, testResult);
+  }
+
+  async onEnd(): Promise<void> {
+    await this.awaitForRunId().catch(e => {
+      this.log(colors.red(e));
+    });
+    if (this.isDisabled)
+      return;
+
+    let prevCount = this.resultsToBePublished.length;
+    while (this.resultsToBePublished.length > 0) {
+      // need wait all results to be published
+      if (prevCount > this.resultsToBePublished.length) {
+        this.log(
+            colors.gray(`Waiting for all results to be published. Remaining ${this.resultsToBePublished.length} results`)
+        );
+        prevCount--;
+      }
+      await new Promise(resolve => setTimeout(resolve, 250));
+    }
+
+    if (this.publishedResultsCount === 0 && !this.runId) {
+      this.log('No testcases were matched. Ensure that your tests are declared correctly.');
+      return;
+    }
+
+    try {
+      const runUpdate: TestInterfaces.RunUpdateModel = {
```

Comment:
it's not used anywhere else, I'd put it right into line 230, instead of storing it inside a variable.